cli/update.go

```diff
@@ -0,0 +1,227 @@
+package cli
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/configs"
+	"github.com/bitrise-io/bitrise/plugins"
+	"github.com/bitrise-io/bitrise/version"
+	"github.com/bitrise-io/go-utils/log"
+	ver "github.com/hashicorp/go-version"
+	"github.com/urfave/cli"
+)
+
+const (
+	tagsURL     = "https://api.github.com/repos/bitrise-io/bitrise/tags"
+	downloadURL = "https://github.com/bitrise-io/bitrise/releases/download/%s/bitrise-%s-x86_64"
+)
+
+var updateCommand = cli.Command{
+	Name:  "update",
+	Usage: "Updates the Bitrise CLI.",
+	Action: func(c *cli.Context) error {
+		log.Infof("Updating Bitrise CLI...")
+		version := c.String("version")
+
+		withBrew, err := installedWithBrew()
+		if err != nil {
+			return err
+		}
+		if withBrew {
+			if version != "" {
+				return errors.New("it seems you installed Bitrise CLI with Homebrew. Version flag is only valid for GitHub release page installations")
+			}
+			cmd := exec.Command("brew", "upgrade", "bitrise")
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			return cmd.Run()
+		}
+
+		if version == "" {
+			latest, err := latestTag()
+			if err != nil {
+				return err
+			}
+			version = latest.String()
+		}
+
+		path, err := exec.LookPath(os.Args[0])
```

Comment:
on linux first you need to unlink the current running bin: https://stackoverflow.com/a/1712109